modules/ROOT/pages/dataweave-cookbook-flat-file-read-and-write.adoc

```diff
@@ -0,0 +1,282 @@
+
+The following examples show you how to read the contents of a flat file, and also how to write a flat file following the specified schema. Each example consists of a Mule application that uses DataWeave to read and write flat files.
+
+When you work with flat files, configure the flat file schema to use for either the write or read operations. The flat file schema defines the structure of a flat file, and this enables Mule to write and read in such format.
+
+These example applications are configured to work with the sample flat file data and the example flat file schema provided in xref:dataweave-flat-file-schemas#full_schema_ex[Flat File Schemas].
+
+== Example of Reading a Flat File
+
+The following example application consists of:
+
+. A Scheduler component that triggers the flow
+. A File Read operation that reads a flat file from the specified path
+. A Transform component that transforms the content read from the flat file to JSON format
+. A Logger component that outputs the current payload
+
+image:mruntime-read-flat-file.png[Example Mule application]
+
+.Application XML File
+[source,xml,linenums]
+----
+<mule xmlns:file="http://www.mulesoft.org/schema/mule/file"
+  xmlns:ee="http://www.mulesoft.org/schema/mule/ee/core"
+  xmlns="http://www.mulesoft.org/schema/mule/core"
+  xmlns:doc="http://www.mulesoft.org/schema/mule/documentation"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/ee/core/current/mule-ee.xsd
+http://www.mulesoft.org/schema/mule/file http://www.mulesoft.org/schema/mule/file/current/mule-file.xsd">
+  <file:config name="File_Config" doc:name="File Config" />
+  <configuration-properties doc:name="Configuration properties" file="flatfileconfig.properties" />
+  <flow name="ReadFlatfileFlow" >
+    <scheduler doc:name="Scheduler">
+      <scheduling-strategy>
+        <fixed-frequency frequency="10000" />
+      </scheduling-strategy>
+    </scheduler>
+    <file:read doc:name="Read flat file" config-ref="File_Config"
+    path="${fileReadPath}"
+    outputMimeType='application/flatfile; schemapath=${schemaPath}'/> //<1>
+    <ee:transform doc:name="Transform to JSON"  >
+      <ee:message >
+        <ee:set-payload ><![CDATA[%dw 2.0
+        output application/json
+        ---
+        payload]]>
+        </ee:set-payload>
+      </ee:message>
+    </ee:transform>
+    <logger level="INFO" doc:name="Logger" message='#[payload]' />
+  </flow>
+</mule>
+----
+
+<1> Note that the File Read operation configures the following attributes:
++
+* `path`: The path of the flat file to read
+* `outputMimeType`: The output MIME type of the content, in this case `application/flatfile`
+* `schemapath`: The reader property that specifies the path to the flat file schema (`.ffs` file)
+
+=== Transformation Output
```

Comment:
I wonder if it is clear how this subtopic relates to the previous container?
I don't know...I feel like we might need some "input" subhead to be a peer to this "output" title?
Also, can we have just a bit of text here to transition from title to syntax?

Reply:
Oh! Are you using a title here to work as a head? 
I believe in semantic tagging: tagging for what it is rather than how it looks, so...I'd be tempted to use a sentence here instead of a heading.